src/components/pages/Home/Onboarding/Steps/ClaimTokens.tsx

```diff
@@ -0,0 +1,157 @@
+
+const query = graphql`
+  query ClaimTokensQuery {
+    file(
+      relativePath: { eq: "pages/index/onboarding/steps/claimTokens.json" }
+    ) {
+      childStepsJson {
+        title
+        subtitle
+        body
+        image {
+          childImageSharp {
+            original {
+              src
+            }
+          }
+        }
+        gxButtonLabel
+        existingGxBalance
+        gxSuccess
+        oceanButtonLabel
+        existingOceanBalance
+        oceanSuccess
+      }
+    }
+  }
+`
+
+type ClaimTokensStep<T> = Partial<T> & {
+  gxButtonLabel: string
+  existingGxBalance: string
+  gxSuccess: string
+  oceanButtonLabel: string
+  existingOceanBalance: string
+  oceanSuccess: string
+}
+
+enum Tokens {
+  GX = 'gx',
+  OCEAN = 'ocean'
+}
+
+export default function ClaimTokens(): ReactElement {
+  const data = useStaticQuery(query)
+  const {
+    title,
+    subtitle,
+    body,
+    image,
+    gxButtonLabel,
+    existingGxBalance,
+    gxSuccess,
+    oceanButtonLabel,
+    existingOceanBalance,
+    oceanSuccess
+  }: ClaimTokensStep<OnboardingStep> = data.file.childStepsJson
+
+  const { accountId, balance, networkId, web3Provider } = useWeb3()
+  const [tokenState, setTokenState] = useState({
+    [Tokens.GX]: { loading: false, touched: false, completed: false },
+    [Tokens.OCEAN]: { loading: false, touched: false, completed: false }
+  })
+
+  useEffect(() => {
+    if (networkId !== GX_NETWORK_ID) {
+      setTokenState({
+        [Tokens.GX]: { ...tokenState.gx, completed: false },
+        [Tokens.OCEAN]: { ...tokenState.ocean, completed: false }
+      })
+      return
+    }
+
+    setTokenState({
+      [Tokens.GX]: { ...tokenState.gx, completed: Number(balance?.eth) > 0 },
+      [Tokens.OCEAN]: {
+        ...tokenState.ocean,
+        completed: Number(balance?.ocean) > 0
+      }
+    })
+  }, [accountId, balance, networkId])
+
+  const claimTokens = async (address: string, token: Tokens) => {
+    if (networkId !== GX_NETWORK_ID) {
+      toast.error(
+        getErrorMessage({
+          accountId,
+          web3Provider: !!web3Provider,
+          networkId,
+          balance: null
+        })
+      )
+    }
+
+    setTokenState({
+      ...tokenState,
+      [token]: { ...tokenState[token], loading: true, touched: true }
+    })
+    const baseUrl =
+      token === Tokens.GX
+        ? 'https://faucet.gx.gaiaxtestnet.oceanprotocol.com/send'
+        : 'https://faucet.gaiaxtestnet.oceanprotocol.com/send'
+
+    try {
+      await axios.get(baseUrl, {
+        params: { address }
+      })
+    } catch {
+      // Workaround until we deploy our own faucet:
+      // the api call is going to fail due to a CORS error but the tokens are
+      // sent anyway so we set the new token state in the catch
+      setTokenState({
+        ...tokenState,
+        [token]: {
+          ...tokenState[token],
+          completed: true,
+          touched: true,
+          loading: false
+        }
```

Comment:
issue(blocking): How do you deal with the case if someone already claimed the token for that day? Is this ignored for v1?

Reply:
Right now we are not showing the button to claim tokens at all if you already have some tokens in your wallet, this is also the behaviour of the faucet as far as I know. 
Also being this tool aimed for the first onboarding it's unlikely the case of a user who got some tokens from the faucet, spent them all and after that came to the portal to get more through our guided process.
We can probably handle this easily when we'll have our own implementation of the faucet but I personally think it's out of scope for this v1.

Reply:
Sounds good enough for v1. Please create an issue so that we can tackle this in our next release.

Reply:
Done: #120